core/src/test/scala/integration/kafka/api/SaslSslAdminIntegrationTest.scala

```diff
@@ -331,6 +332,25 @@ class SaslSslAdminIntegrationTest extends BaseAdminIntegrationTest with SaslSetu
     assertFutureExceptionTypeEquals(results.values.get(emptyResourceNameAcl), classOf[InvalidRequestException])
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testCreateDelegationTokenWithOptions(quorum: String): Unit = {
+    client = createAdminClient
+    val timeout = 1000
+    val options = new CreateDelegationTokenOptions().maxlifeTimeMs(timeout);
+    val token = client.createDelegationToken(options).delegationToken().get()
+    assertEquals(timeout, token.tokenInfo.maxTimestamp - token.tokenInfo.issueTimestamp)
+  }
```

Comment:
According to the test case, I think this test should cover all options, but only the `maxlifeTimeMs` option is being tested.
Could you test the rest of them ?